src/v/pandaproxy/schema_registry/sharded_store.cc

```diff
@@ -241,9 +242,11 @@ ss::future<bool> sharded_store::upsert(
   schema_id id,
   schema_version version,
   is_deleted deleted) {
+    auto norm = normalize{
+      config::shard_local_cfg().schema_registry_normalize_on_startup()};
     co_return co_await upsert(
       marker,
-      co_await make_canonical_schema(std::move(schema)),
+      co_await make_canonical_schema(std::move(schema), norm),
```

Comment:
Are there any assumptions in the SR code that the schema stored in-memory is the same as the one on the topic? I'm wondering if `norm=true` here could cause any unexpected behaviour.

Reply:
> Are there any assumptions in the SR code that the schema stored in-memory is the same as the one on the topic? I'm wondering if `norm=true` here could cause any unexpected behaviour.

No, it always goes through canonicalization, and optionally normalization.